library/src/main/kotlin/com/chuckerteam/chucker/api/entity/ManualHttpTransaction.kt

```diff
@@ -0,0 +1,74 @@
+package com.chuckerteam.chucker.api.entity
+
+import com.chuckerteam.chucker.internal.data.entity.HttpTransaction
+
+/**
+ * Represent Manual Http transaction that developer want to populate as an custom http transaction
+ */
+public data class ManualHttpTransaction(
```

Comment:
Why do we need another class here?

Reply:
I didn't want to move `HttpTransaction` class to the API package or expose it to the developer. I could use a builder pattern to build transaction objects or create a method with a bunch of args in ChuckerCollector class.
but both of these mentioned approaches has it downsides regarding logging the data. In this implemented approach, to be able to log the start and end information of a transaction, the developer no longer needs to store the data in a separate data store and can create the model and fill it in during transactions. at the end send it to be stored in the chucker.